From 7382af7ba92102f4d4191cae8dc547d02305ed9a Mon Sep 17 00:00:00 2001
From: Amit Kumar Mahapatra <amit.kumar-mahapatra@xilinx.com>
Date: Wed, 6 May 2020 00:01:28 +0530
Subject: [PATCH] mtd: spi-nor: Add 4byte opcode support for Macronix flash
 parts

Added 4byte opcode support for MX66L2G45G and MX66U2G45G Macronix flash parts.

Signed-off-by: Amit Kumar Mahapatra <amit.kumar-mahapatra@xilinx.com>
---
 drivers/mtd/spi-nor/spi-nor.c | 6 ++++--
 1 file changed, 4 insertions(+), 2 deletions(-)

diff --git a/drivers/mtd/spi-nor/spi-nor.c b/drivers/mtd/spi-nor/spi-nor.c
index 51265c1bd72b..7b407e32252e 100644
--- a/drivers/mtd/spi-nor/spi-nor.c
+++ b/drivers/mtd/spi-nor/spi-nor.c
@@ -2628,9 +2628,11 @@ static const struct flash_info spi_nor_ids[] = {
 	{ "mx66u1g45g",  INFO(0xc2253b, 0, 64 * 1024, 2048, SECT_4K |
 			SPI_NOR_DUAL_READ | SPI_NOR_QUAD_READ) },
 	{ "mx66l2g45g",  INFO(0xc2201c, 0, 64 * 1024, 4096, SECT_4K |
-			SPI_NOR_DUAL_READ | SPI_NOR_QUAD_READ) },
+			SPI_NOR_DUAL_READ | SPI_NOR_QUAD_READ |
+			SPI_NOR_4B_OPCODES) },
 	{ "mx66u2g45g",  INFO(0xc2253c, 0, 64 * 1024, 4096, SECT_4K |
-			SPI_NOR_DUAL_READ | SPI_NOR_QUAD_READ) },
+			SPI_NOR_DUAL_READ | SPI_NOR_QUAD_READ |
+			SPI_NOR_4B_OPCODES) },
 	{ "mx66l1g55g",  INFO(0xc2261b, 0, 64 * 1024, 2048, SPI_NOR_QUAD_READ) },
 
 	/* Micron <--> ST Micro */
-- 
2.17.1

